File: python/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        right_sum = sum(nums)
        left_sum = 0
        for i in range(len(nums)):
            right_sum -= nums[i]
            if right_sum == left_sum:
                return i
            left_sum += nums[i]
        return -1

File: rust/find-pivot-index.rs
impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut total_sum: i32 = nums.iter().sum();
        let mut current_sum: i32 = 0;
        for i in 0..nums.len() {
            total_sum -= nums[i];
            if current_sum == (total_sum) {
                return i as i32;
            }
            current_sum += nums[i];
        }
        return -1;
    }
}
